### PR TITLE
Adds hygiene and nutrition scan options to scanner gates

### PR DESCRIPTION
**Pull request by 81Denton**
:cl: Denton
tweak: Scanner gates can now check for hygiene levels (dirty or clean) and nutrition levels (starving or obese).
/:cl:

This PR adds a hygiene scan option to alert you whenever stinky (or clean) people pass through it. Great for keeping track of assistants.
Edit: Also, nutrition modes! You can select either starving or obese nutrition levels.

### CODE REVIEW

**Comment by PKPenguin321** [comment]
Brappers-only clubs version of metal detectors

**Comment by 81Denton** [comment]
Should I also add nutrition scanning (eg starving/fat)?

**Comment by Bawhoppen** [comment]
@81Denton I would say big yes, but in another PR.

**Comment by ExcessiveUseOfCobblestone** [comment]
You can add both if you'd like here.

File: code/game/machinery/scan_gate.dm [hunk 1]
```diff
@@ -115,6 +117,19 @@
 				if(istype(I, /obj/item/gun))
 					beep = TRUE
 					break
+		if(SCANGATE_HYGIENE)
+			if(iscarbon(M))
+				var/mob/living/carbon/human/H = M
+				if(H.hygiene <= HYGIENE_LEVEL_DIRTY) //Pungent indeed
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/machinery/scan_gate.dm]: you should have the invert option as well.

We can have High-class areas and "Neet Meet N' Greet" areas

Comment (ExcessiveUseOfCobblestone) [reply code/game/machinery/scan_gate.dm]: Reading code in 2019 hah!

**Comment by Anonmare** [comment]
Time to make a bar that has banned fat chicks

File: code/game/machinery/scan_gate.dm [hunk 2]
```diff
@@ -115,6 +118,21 @@
 				if(istype(I, /obj/item/gun))
 					beep = TRUE
 					break
+		if(SCANGATE_HYGIENE)
+			if(iscarbon(M))
+				var/mob/living/carbon/human/H = M
```
Comment (Qustinnus) [inline code/game/machinery/scan_gate.dm]: you checked if its carbon, not if its human

File: code/game/machinery/scan_gate.dm [hunk 3]
```diff
@@ -115,6 +118,21 @@
...
+				if(H.hygiene >= detect_hygiene && detect_hygiene == HYGIENE_LEVEL_CLEAN)
+					beep = TRUE
+				if(H.hygiene <= detect_hygiene && detect_hygiene == HYGIENE_LEVEL_DIRTY)
+					beep = TRUE
+		if(SCANGATE_NUTRITION)
+			if(iscarbon(M))
```
Comment (Qustinnus) [inline code/game/machinery/scan_gate.dm]: ditto

File: code/game/machinery/scan_gate.dm (hunk 2 above)
Comment (81Denton) [reply code/game/machinery/scan_gate.dm]: time 4 webedits

**Pull request merged**